jass/lib/UnitDex.j

```diff
@@ -0,0 +1,434 @@
+library UnitDex uses optional WorldBounds, optional GroupUtils
+/***************************************************************
+*
+*   v1.2.1, by TriggerHappy
+*   ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+*   UnitDex assigns every unit an unique integer. It attempts to make that number within the
+*   maximum array limit so you can associate it with one.
+*   _________________________________________________________________________
+*   1. Installation
+*   ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+*   Copy the script to your map, save it, then restart the editor and comment out the line below.
+*/
+    ///! external ObjectMerger w3a Adef uDex anam "Detect Leave" ansf "(UnitDex)" aart "" acat "" arac 0
```

Comment:
We don't have `ObjectMerger` because we run in `--scriptonly `mode, I guess we'll have to import this line to the map, save it with Newgen, then remove this line? Alternatively we could create this exact object using wurst object generation which has a much nicer interface.